jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridPresenter.java

```diff
@@ -162,70 +178,179 @@ public void getData( final Range visibleRange ) {
-                                            dataSetQueryHelper.getColumnLongValue( dataSet, DataSetTasksListGridViewImpl.COLUMN_PROCESSSESSIONID, i ),
-                                            dataSetQueryHelper.getColumnLongValue( dataSet, DataSetTasksListGridViewImpl.COLUMN_PROCESSINSTANCEID, i ),
-                                            dataSetQueryHelper.getColumnStringValue( dataSet, DataSetTasksListGridViewImpl.COLUMN_DEPLOYMENTID, i ),
-                                            dataSetQueryHelper.getColumnLongValue( dataSet, DataSetTasksListGridViewImpl.COLUMN_PARENTID, i ) ) );
-
-                                }
-                                PageResponse<TaskSummary> taskSummaryPageResponse = new PageResponse<TaskSummary>();
-                                taskSummaryPageResponse.setPageRowList( myTasksFromDataSet );
-                                taskSummaryPageResponse.setStartRowIndex( visibleRange.getStart() );
-                                taskSummaryPageResponse.setTotalRowSize( dataSet.getRowCountNonTrimmed() );
-                                taskSummaryPageResponse.setTotalRowSizeExact( true );
-                                if ( visibleRange.getStart() + dataSet.getRowCount() == dataSet.getRowCountNonTrimmed() ) {
-                                    taskSummaryPageResponse.setLastPage( true );
-                                } else {
-                                    taskSummaryPageResponse.setLastPage( false );
-                                }
-                                DataSetTasksListGridPresenter.this.updateDataOnCallback( taskSummaryPageResponse );
-                            }
-                            view.hideBusyIndicator();
-                        }
+                    dataSetQueryHelper.setDataSetHandler(currentTableSettings);
+                    dataSetQueryHelper.lookupDataSet(visibleRange.getStart(), createDataSetTaskCallback(visibleRange.getStart(), currentTableSettings));
+
+                }
+            }
+        } catch ( Exception e ) {
+        }
+
+    }
+
+    protected DataSetReadyCallback createDataSetTaskCallback(final int startRange, final FilterSettings tableSettings){
+        return new DataSetReadyCallback() {
+
+
+            @Override
+            public void notFound() {
+                view.hideBusyIndicator();
+                errorPopup.showMessage("Not found DataSet with UUID [  jbpmHumanTasks ] ");
+            }
 
-                        @Override
-                        public void notFound() {
-                            view.hideBusyIndicator();
-                            errorPopup.showMessage( "Not found DataSet with UUID [  jbpmHumanTasks ] " );
-                            GWT.log( "DataSet with UUID [  jbpmHumanTasks ] not found." );
+            @Override
+            public boolean onError( final ClientRuntimeError error ) {
+                view.hideBusyIndicator();
+                error.getThrowable().printStackTrace();
```

Comment:
For this one, I think using GWT.log("some error message", error.getThrowable()); would probably be a better way to handle it.